File: app/src/main/java/net/crunkhouse/shaitzov/ShaitzovApplication.java
package net.crunkhouse.shaitzov;

import android.app.Application;

import timber.log.Timber;

/**
 * Application class to set up some things on app startup.
 */

public class ShaitzovApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        LocalPreferences.init(getApplicationContext());
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }
    }
}

File: app/src/main/java/net/crunkhouse/shaitzov/ShaitzovApplication.kt
package net.crunkhouse.shaitzov

import android.app.Application

import timber.log.Timber

/**
 * Application class to set up some things on app startup.
 */

class ShaitzovApplication : Application() {

    lateinit var prefs: LocalPreferences

    override fun onCreate() {
        super.onCreate()
        prefs = LocalPreferences(applicationContext)
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
    }
}
